apps/ledger-live-mobile/src/components/Web3AppWebview/PlatformAPIWebview.tsx

```diff
@@ -58,6 +70,8 @@ function renderLoading() {
 }
 export const PlatformAPIWebview = forwardRef<WebviewAPI, WebviewProps>(
   ({ manifest, inputs = {}, onStateChange }, ref) => {
+    const isNativeSwap = useMemo(() => currentRouteNameRef.current === "Swap", []);
```

Comment:
```suggestion
    const tracking =  trackWithNativeOrigin(currentRouteNameRef.current === "Swap")
```

We can close over the ref here, once. If you instead declare `tracking` here to the function, it closes over the ref, and then the rest of this file can remain the same, and look cleaner.  i.e. `tracking().platformRequestAccountRequested(manifest)` can go back to `tracking.platformRequestAccountRequested(manifest)`

Reply:
Here is a test demonstrating what I mean:

```jest
function appManifestFixture(name = "live-app"): AppManifest {
  return {
    id: "12",
    name,
    url: "https://www.ledger.fr",
    homepageUrl: "https://www.ledger.fr",
    platforms: ["ios", "android", "desktop"],
    apiVersion: "1.0.0",
    manifestVersion: "1.0.0",
    branch: "debug",
    categories: ["test"],
    currencies: "*",
    content: {
      shortDescription: {
        en: "test",
      },
      description: {
        en: "test",
      },
    },
    permissions: [],
    domains: [],
    visibility: "complete",
  };
}

describe("tracking() with flowInitiatedFrom ", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  const mockedTrack = jest.fn();
  const manifest = appManifestFixture();
  const trackWithNativeOrigin = (isNativeSwap?: boolean) =>
    trackingWrapper(
      (
        eventName: string,
        properties?: Record<string, unknown> | null,
        mandatory?: boolean | null,
      ) =>
        mockedTrack(
          eventName,
          {
            ...properties,
            flowInitiatedFrom: isNativeSwap ? "Native" : "Discover",
          },
          mandatory,
        ),
    );

  it("adds Discover to the tracking props by default", () => {
    // When
    const tracking = trackWithNativeOrigin();
    tracking.requestAccountRequested(manifest);

    // Then
    expect(mockedTrack).toHaveBeenCalledTimes(1);
    expect(mockedTrack).toHaveBeenCalledWith(
      "WalletAPI RequestAccount",
      {
        walletAPI: manifest.name,
        flowInitiatedFrom: "Discover",
      },
      null,
    );
  });

  it("adds Native when isSwap is true", () => {
    // When
    const tracking = trackWithNativeOrigin(true);
    tracking.requestAccountRequested(manifest);
    // Then
    expect(mockedTrack).toHaveBeenCalledTimes(1);
    expect(mockedTrack).toHaveBeenCalledWith(
      "WalletAPI RequestAccount",
      {
        walletAPI: manifest.name,
        flowInitiatedFrom: "Native",
      },
      null,
    );
  });
});


```

Reply:
Unfortunately this was done this way for a reason.  The component rerenders causing the ref to change and losing the original value we needed so has to be memoised

Reply:
that's weird, refs should not be affected by re-renders! But you can still close over the memoized value, `const tracking = trackWithNativeOrigin(isNativeSwap);`. 

Or just trackWithNativeOrigin(false), in this case, potentially, if @sarneijim Sara is saying that it's never native origin in this webview ..?

Reply:
(I don't know what Native vs Discover origin means)